cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -31,6 +31,7 @@
 // We need BOOST_USE_WINDOWS_H definition with MinGW when we use
 // boost/process.hpp. See BOOST_USE_WINDOWS_H=1 in
 // cpp/cmake_modules/ThirdpartyToolchain.cmake for details.
+#define BOOST_NO_CXX98_FUNCTION_BASE  // ARROW-17805
```

Comment:
Could you move this to `#20` (before `// This boost/asio/...`) to avoid confusing the above `// We need ...` comment target?

Reply:
done